### ISSUE TITLE
Virtual DOM?

### ISSUE THREAD

**Issue by micahlt** [lankybox02/RiverBox#23]
Right now the code is _super_ messy with all of the template strings and innerHTML references.  I get not wanting to use a framework, but what about a Virtual DOM (maybe [million.js](https://github.com/aidenybai/million)?) that would simplify and speed up creating DOM elements?

**Comment by aidenybai** [comment]
> Right now the code is _super_ messy with all of the template strings and innerHTML references.  I get not wanting to use a framework, but what about a Virtual DOM (maybe [million.js](https://github.com/aidenybai/million)?) that would simplify and speed up creating DOM elements?

Happy to help with this is you run into any problems!

**Comment by aidenybai** [comment]
Just got back to this issue. Integration of Million should be pretty simple -- Here are some general refactoring guides:

The `<script>` tag must have a `type="module" attribute` to import Million:

```js
import { render } from 'https://unpkg.com/million';
import { html } from 'https://unpkg.com/million/dist/html.mjs';

const handler = () => alert('Hello! I was clicked!');

// you can pass event listeners / non string data into the html`data`
render(
	document.getElementById('#example-el'), 
	html`<div>Will be put into #example-el <button onclick=${handler}>Hello!</button></div>`
);
```

**Comment by micahlt** [comment]
@lankybox02 what are your thoughts on this issue?

**Comment by lankybox02** [comment]
bruh
![image](https://user-images.githubusercontent.com/104015308/167918499-85e0273c-9eb4-4842-8845-f9faff5a4654.png)
![image](https://user-images.githubusercontent.com/104015308/167918648-053956bc-dff5-475a-896a-67f91ae2ef30.png)

**Comment by aidenybai** [comment]
Can you try using the unpkg version? I dunno if skypack will work correctly

**Comment by lankybox02** [comment]
> Can you try using the unpkg version? I dunno if skypack will work correctly

I did, this is what it returned:
![image](https://user-images.githubusercontent.com/104015308/167932445-f5cc6f8b-238c-418e-a5bb-805a71a53886.png)

**Comment by lankybox02** [comment]
@aidenybai

**Comment by aidenybai** [comment]
Hmm.. are you sure the script tag has a type="module"?

**Comment by lankybox02** [comment]
@aidenybai yes, everything works fine. The render function is being faulty.

**Comment by aidenybai** [comment]
I cannot repro the issue with a basic example. Can you send a screenshot of how you're using it?

**Comment by lankybox02** [comment]
```
import { render } from 'https://unpkg.com/million';
import { html } from 'https://unpkg.com/million/dist/html.mjs';

render(
	document.getElementById('#pageContent'), 
	html`<div>posttest</div>`
);
```

**Comment by aidenybai** [comment]
I'm still not able to repro the issue (demo: https://codesandbox.io/s/festive-surf-1228ym?file=/index.html)

Can you check if the `#pageContent` element exists on the page? Insert this code snippet prior to the render function call:

```js
if (document.body.contains(document.getElementById('#pageContent'))) {
	console.log('exists');
} else {
	console.log('does not exist, cannot render to an element that doesn't exist');
}
```

**Comment by lankybox02** [comment]
@aidenybai it exists

**Comment by micahlt** [comment]
@aidenybai @lankybox02 you're using `document.getElementById` like you would `document.querySelector`.  The issue is the `#` symbol in the function.  In other words, `document.getElementById("#pageContent")` should be just `document.getElementById("pageContent")`.

**Comment by lankybox02** [comment]
Just pushed out beta 1.4, Pull Requests for million.js are open

**Comment by lankybox02** [comment]
I'm going to start working on this in 1.7 or 1.8

**Comment by lankybox02** [comment]
Progress is doing alright so far
